### ISSUE TITLE
[Bug]: Postgres bulkloader throws class loading exception. HOP-4600

### ISSUE THREAD

**Issue by usbrandon** [apache/hop#2056]
### Apache Hop version?

2.2

### Java version?

openjdk version "11.0.17" 2022-10-18

### Operating system

Linux

### What happened?

If you try to run the Postgress Bulk Loader against a table, it immediately throws an exception about casting objects.  

2022/11/16 15:50:34 - PostgreSQL Bulk Loader.0 - ERROR: Error in transform
2022/11/16 15:50:34 - PostgreSQL Bulk Loader.0 - ERROR: org.apache.hop.core.exception.HopException: 
2022/11/16 15:50:34 - PostgreSQL Bulk Loader.0 - Error while preparing the COPY COPY "Production"."fact_sales_analysis" ( "QUANTITY_UNITS", "QUANTITY", "FILL_DATE_CREATED_TK", "FORMULA_TK", "DOCTOR_TK", "PATIENT_TK", "Business_Line", "Invoice_Date_TK", "Invoice_time_TK", "Invoice_No", "Transaction_Type", "Invoice_Desc", "Invoice_Line_Item_No", "Line_Item", "Item_Type", "Rx_No", "Rx_Fill_ID", "Revenue", "CUSTOMER_TK", "SYSTEM_ID" )  FROM STDIN WITH CSV DELIMITER AS ';' QUOTE AS '"';
2022/11/16 15:50:34 - PostgreSQL Bulk Loader.0 - 
2022/11/16 15:50:34 - PostgreSQL Bulk Loader.0 - class org.postgresql.jdbc.PgConnection cannot be cast to class org.postgresql.PGConnection (org.postgresql.jdbc.PgConnection is in unnamed module of loader org.apache.hop.core.plugins.HopURLClassLoader @1462a99d; org.postgresql.PGConnection is in unnamed module of loader org.apache.hop.core.plugins.HopURLClassLoader @15d18b61)
2022/11/16 15:50:34 - PostgreSQL Bulk Loader.0 - 
2022/11/16 15:50:34 - PostgreSQL Bulk Loader.0 -        at org.apache.hop.pipeline.transforms.pgbulkloader.PGBulkLoader.doCopy(PGBulkLoader.java:161)
2022/11/16 15:50:34 - PostgreSQL Bulk Loader.0 -        at org.apache.hop.pipeline.transforms.pgbulkloader.PGBulkLoader.processRow(PGBulkLoader.java:240)
2022/11/16 15:50:34 - PostgreSQL Bulk Loader.0 -        at org.apache.hop.pipeline.transform.RunThread.run(RunThread.java:55)
2022/11/16 15:50:34 - PostgreSQL Bulk Loader.0 -        at java.base/java.lang.Thread.run(Thread.java:829)
2022/11/16 15:50:34 - PostgreSQL Bulk Loader.0 - Caused by: java.lang.ClassCastException: class org.postgresql.jdbc.PgConnection cannot be cast to class org.postgresql.PGConnection (org.postgresql.jdbc.PgConnection is in unnamed module of loader org.apache.hop.core.plugins.HopURLClassLoader @1462a99d; org.postgresql.PGConnection is in unnamed module of loader org.apache.hop.core.plugins.HopURLClassLoader @15d18b61)
2022/11/16 15:50:34 - PostgreSQL Bulk Loader.0 -        at org.apache.hop.pipeline.transforms.pgbulkloader.PGBulkLoader.doCopy(PGBulkLoader.java:158)
2022/11/16 15:50:34 - PostgreSQL Bulk Loader.0 -        ... 3 more
2022/11/16 15:50:34 - moveProductionFromMSSQLtoPostgres - Pipeline detected one or more transforms with errors.
2022/11/16 15:50:34 - moveProductionFromMSSQLtoPostgres - Pipeline is killing the other transforms!

### Issue Priority

Priority: 3

### Issue Component

Component: Transforms

**Comment by hansva** [comment]
Could it be that you have 2 JDBC drivers in your Hop installation? on in Lib and one in plugins/databases/postgresql/lib ?

**Comment by usbrandon** [comment]
No chance.  I only have one in an external folder.  Each time I try a new snapshot of Hop I delete the postgres jar and mssqlnative jar that it ships with because I have newer JDBC drivers.

**Comment by usbrandon** [comment]
It turns out that the bulkloader works fine.  There is a different bug in Hop related to classloading where if you use JDBC drivers in an external folder then something bad happens and these kinds of errors appear.  The errors go away on the out of the box Hop as it doesn't try to use external JDBC driver folders.  The safest place I found for JDBC jar files is hop/lib/core.   This was what Bart and I discovered troubleshooting my environment.  It caused errors to come up when I tried the Vertica bulkloader.

There is, however, still a null pointer exception when clicking the SQL button on the Postgres Bulkloader transform, but at least the data transfer functionality works.
